packages/frontend/src/components/Game/GameBoard/index.jsx

```diff
@@ -1,17 +1,26 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import useGameContext from 'hooks/useGameContext';
 import ScoreBoard from '../ScoreBoard';
 import TileGrid from './TileGrid';
-
+import { handleGameOver } from 'contexts/game/actions';
+import ResultDisplay from './ResultDisplay';
 const GameBoard = () => {
   const {
-    state: { players, currentPlayer, tiles },
+    state: { players, currentPlayer, tiles, stage },
+    dispatch,
   } = useGameContext();
 
+  useEffect(() => {
+    const matchedTiles = tiles.filter((tile) => tile.isMatched === true);
+    if (tiles.length !== 0 && matchedTiles.length === tiles.length) {
+      dispatch(handleGameOver());
+    }
+  }, [tiles]);
+
   return (
     <>
       <ScoreBoard players={players} currentPlayer={currentPlayer} />
-      <TileGrid tiles={tiles} />
+      {stage === 'PLAYING' ? <TileGrid tiles={tiles} /> : <ResultDisplay />}
```

Comment:
The stage could be pulled from the `GAME_STAGES` object and used in exactly the same way as the string version. `const { PLAYING } = GAME_STAGES` and then `stage === PLAYING`, without quotes